packages/flutter/test/material/banner_test.dart

```diff
@@ -1105,6 +1105,27 @@ void main() {
       ),
     );
   });
+
+   testWidgets('Custom Margin respected', (WidgetTester tester) async {
+    const EdgeInsets margin = EdgeInsets.all(30);
+    await tester.pumpWidget(
+      MaterialApp(
+        home: MaterialBanner(
+         margin: margin,
+          content: const Text('I am a banner'),
+          actions: <Widget>[
+            TextButton(
+              child: const Text('Action'),
+              onPressed: () { },
+            ),
+          ],
+        ),
+      ),
+    );
+
+    final Offset topLeft = tester.getTopLeft(find.descendant(of: find.byType(MaterialBanner), matching: find.byType(Material)).first);
```

Comment:
I have used the same approach that other tests are using in `banner_test.dart`, Getting Material from the banner and checking the offset of the material.